### PR TITLE
chore: [SIW-1091] Upgrade react-native to 0.72.14

### PR DESCRIPTION
**Pull request by gispada**
#### List of Changes

This PR updates React Native to version 0.72.14 in both the project and the example application.

In the newer version, `@react-native-community/eslint-config` has been replaced by `@react-native/eslint-config`. A few gems were also updated.

**Note:** this PR supersedes #75.

#### How Has This Been Tested?

The example app has been run on the iOS Simulator and the Android Emulator.

### CODE REVIEW

File: example/ios/PrivacyInfo.xcprivacy
Comment (hevelius) [inline example/ios/PrivacyInfo.xcprivacy:1]: What do you think to move this file under example/ios/IoReactNativeWalletExample? The same folder of Info.plist

File: example/ios/Podfile.lock [hunk 1]
```diff
@@ -70,9 +70,9 @@ PODS:
     - FlipperKit/FlipperKitNetworkPlugin
   - fmt (6.2.1)
   - glog (0.3.5)
-  - hermes-engine (0.72.4):
-    - hermes-engine/Pre-built (= 0.72.4)
-  - hermes-engine/Pre-built (0.72.4)
+  - hermes-engine (0.72.12):
+    - hermes-engine/Pre-built (= 0.72.12)
+  - hermes-engine/Pre-built (0.72.12)
```
Comment (hevelius) [inline example/ios/Podfile.lock:75]: Why hermes-engine is 0.72.12 instead 0.72.14? Have you run a `pod install` after upgrade?

**Review by hevelius — CHANGES_REQUESTED** [review]

File: example/ios/Podfile.lock (hunk 1 above)
Comment (gispada) [reply example/ios/Podfile.lock]: Good catch, I run it but somehow hermes-engine did not get updated.

**Review by hevelius — APPROVED** [review]
LGTM

**Pull request merged**